### ISSUE TITLE
Installation of ops-agent for fleet of VMs

### ISSUE THREAD

**Issue by DeepaMadheswaran** [terraform-google-modules/terraform-google-cloud-operations#72]
### TL;DR

We are trying to install/enable the ops-agent for fleet of VMs. 
  - We have tried by providing the terraform script
  
  module "agent_policy" {
  source     = "terraform-google-modules/cloud-operations/google//modules/agent-policy"
  version    = "~> 0.2.3"
  project_id = "dataworkz-devops"
  policy_id  = "ops-agents-example-policy"
  agent_rules = [
    {
      type               = "ops-agent"
      version            = "current-major"
      package_state      = "installed"
      enable_autoupgrade = true
    },
  ]
  os_types = [
    {
      short_name = "ubuntu"
      version    = "22.04"
    },
  ]
}
terraform apply show that resource is created, but in Gcloud console shows that ops-agent not installed and also metrics are not uploading

### Expected behavior

Metrics should be to uploaded and viewable from GCloud console metrics explorer for the GKE node group VMs

### Observed behavior

Ops-agent policy is applied, but ops-agent is not installed in the node group VMs 

### Terraform Configuration

```hcl
module "agent_policy" {
  source     = "terraform-google-modules/cloud-operations/google//modules/agent-policy"
  version    = "~> 0.2.3"
  project_id = "dataworkz-devops"
  policy_id  = "ops-agents-example-policy"
  agent_rules = [
    {
      type               = "ops-agent"
      version            = "current-major"
      package_state      = "installed"
      enable_autoupgrade = true
    },
  ]
  os_types = [
    {
      short_name = "ubuntu"
      version    = "22.04"
    },
  ]
}
```

### Terraform Version

```sh
Terraform v1.3.5
```

**Comment by dgulli** [comment]
also getting this, same problem.

**Comment by cyanezp** [comment]
I was facing the same problems in my project: the policy is created correctly but nothing happens.
It turns out that the problem is not related to this module itself, but to the requirements and extra configurations that must be carried out for the policy to take effect.

## Key points:

- First of all, make sure that your project has the OS Config API (osconfig.googleapis.com) enabled.
- Your VMs must have the following metadata, which can be created directly on your compute instance or on your instance template:
```terraform
metadata = {
    enable-osconfig = "TRUE"
    enable-guest-attributes = "TRUE"
}
```

- Each VM must have attached a service account with the necessary privileges to send the information collected by the OS agent:
```terraform
service_account {
    scopes = [
      "https://www.googleapis.com/auth/devstorage.read_only",
      "https://www.googleapis.com/auth/logging.write",
      "https://www.googleapis.com/auth/monitoring.write",
      "https://www.googleapis.com/auth/pubsub",
      "https://www.googleapis.com/auth/service.management.readonly",
      "https://www.googleapis.com/auth/servicecontrol",
      "https://www.googleapis.com/auth/trace.append",
    ]
  }
```

## Related resources:
This resources helped me out to find out what was wrong on my project:

- https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/managing-agent-policies
- https://cloud.google.com/compute/docs/manage-os#debian
- https://cloud.google.com/compute/docs/troubleshooting/vm-manager/verify-setup

Hope this helps

**Comment by geekette86** [comment]
I reopen this because i have all the necessary scopes and metadata but still can't see that the ops agent is installed

**Comment by geekette86** [comment]
Any updates about this  bug
